Mention alternative quantize syntax (and clean-up metrics operators docs)

aszczepaniksumo opened the pull request:
## Purpose of this pull request
This PR adds a mention alternate syntax of `quantize` operator without the `to INTERVAL` section. We do not mention that we support it right now. Request: https://github.com/SumoLogic/sumologic-documentation/issues/2218

I wanted only to update the quantization doc, but noticed that there are several (seriously worrying) inconsistencies in metrics operator docs:
* Weird `metric query |` prefixes in what should be formal syntax specification.
* Use of both block quotes and single ticked strings to define the syntax.
* Inconsistent section names.
* Some other minor typoos, formatting errors, etc

I decided to fix all of those as well.

I recommend to review commit by commit.

## Select the type of change:

- [ ] Minor Changes - Typos, formatting, slight revisions
- [X] Update Content - Revisions and updating sections
- [ ] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - Updates, maintenance, and new packages for the site, Gatsby, React, etc

michaljmatusiak92 commented on docs/metrics/metrics-operators/eval.md:
````diff
@@ -34,15 +34,15 @@ metrics query | eval expr([REDUCER BOOLEAN EXPRESSION | _value] [_granularity])
 
 ```
-_sourceCategory=ApacheHttpServer metrics=cpu_idle | eval _value * 100
+_sourceCategory=ApacheHttpServer metric=cpu_idle | eval _value * 100
````
great catch!

michaljmatusiak92 commented on docs/metrics/metrics-operators/eval.md:
````diff
@@ -34,15 +34,15 @@ metrics query | eval expr([REDUCER BOOLEAN EXPRESSION | _value] [_granularity])
 
 ```
-metrics query | eval avg
+metric=cpu | eval avg
````
It would be good to unify the selector between examples, we have `metric=cpu_idle` and `metric=cpu_idle` , sumo host metrics have `metric=CPU_Idle` and we might want to be consistent with it

michaljmatusiak92 commented on docs/metrics/metrics-operators/ewma.md:
```diff
@@ -19,44 +19,25 @@ If you run `ewma` without specifying either `alpha` or `span`, it runs by defaul
-* `alpha`, the smoothing parameter, is a decimal value (0.0 ≤ alpha ≤ 1.0)
-* The default value of `alpha` is 0.5
+* `alpha`, the smoothing parameter, is a decimal value (0.0 ≤ alpha ≤ 1.0) The default value of `alpha` is 0.5
+* `span` is the number of data points. Must be an integer value greater than zero. If you set `span=5`, the last five data points will be used to calculate the average.  The default value of `span` is 3.
```
`Must be a positive integer` ?

michaljmatusiak92 commented on docs/metrics/metrics-operators/quantize.md:
```diff
@@ -13,12 +13,16 @@ You can specify:
-:::note
-The `quantize` operator *must* appear immediatdly after your query selector, before any other operators.
-:::
+There are two supported syntaxes for the `quantize` operator.
```
just double-check: `quantize to INTERVAL` is not supported? (we might want also to explicitly mention it)

aszczepaniksumo replied:
`quantize to INTERVAL`  is supported. I guess that should be self-explanatory from first syntax:
```
quantize to INTERVAL [using ROLLUP] [drop last]
```
where `[]` denotes optional stuff. We also have an corresponding example.

aszczepaniksumo replied:
A follow up might be: why not to stuff both syntaxes in a single declaration. That would be possible with  some `[]` kung-fu, but I decided that such declaration would be too complex - it's much cleaner to declare two versions.

michaljmatusiak92 replied:
ok, please ignore my comment, the proposal change looks ok ;)

betenkowski replied:
In reality, there are three versions:
1. `quantize to $interval`
2. `quantize using $add`
3. `quantize to $interval using $agg`

Saying there are two versions is even more misleading than `[]` kung-fu in my opinion. I'd write that we need to specify at least one of the two.

michaljmatusiak92 replied:
Agree, my comment about not supporting `quantize to INTERVAL` was my first thought after I've read that we support 2 syntaxes

aszczepaniksumo replied:
I updated it to only one syntax definition and added a note about at least one of `to $interval` `using #rollup` being mandatory. Thanks for a good suggestion. 
Please check if it's clear for you now.

michaljmatusiak92 reviewed (commented):
lgtm in general, added minor comments

michaljmatusiak92 commented on docs/metrics/metrics-operators/min.md:
```diff
@@ -18,17 +18,17 @@ min [by FIELD [, FIELD, ...]]
 ### Minimum metric value across by a tag value
 
-This query returns the minimum cpu_system value by node for each time interval across all matching time series.
+This query returns the minimum `CPU_System` value by node for each time interval across all matching time series.
```
I would suggest to be consistent with `CPU_Sys` and ``CPU_System` as well

aszczepaniksumo replied:
Ops, missed this one. Corrected.

betenkowski commented on docs/metrics/metrics-operators/along.md:
```diff
@@ -10,7 +10,7 @@ keywords:
 import useBaseUrl from '@docusaurus/useBaseUrl';
 
-The `along metrics` operator is useful when you join queries – it allows you to control what results are joined based on the value of one or more result fields. For more information, see [Join Query Results](docs/metrics/introduction/joins.md).
+The `along` metrics operator is useful when you join queries – it allows you to control what results are joined based on the value of one or more result fields. For more information, see [Join Query Results](docs/metrics/introduction/joins.md).
```
Formally, `along` is not an operator, it's just a statement.

kimsauce replied:
@betenkowski can you help us re-word that paragraph?

betenkowski replied:
*The `along` statement is useful...* will work.

betenkowski commented on docs/metrics/metrics-operators/quantize.md:
```diff
@@ -11,14 +11,18 @@ You can specify:
+## Syntax
 
-:::note
-The `quantize` operator *must* appear immediatdly after your query selector, before any other operators.
```
Most likely, this was just an (over)simplification. To apply the quantization directly to the selector, `quantize` has to follow the selector immediately. Otherwise, it will just work on top of whatever has been returned by the previous step.

aszczepaniksumo replied:
I added your comment as a note 👍

betenkowski reviewed (commented):
Left a couple of suggestions. See also https://help.sumologic.com/docs/metrics/introduction/metric-quantization/, maybe we should add more examples there.

aszczepaniksumo commented:
> Left a couple of suggestions. See also https://help.sumologic.com/docs/metrics/introduction/metric-quantization/, maybe we should add more examples there.

I refactored the document slightly and fixed some minor issues. I added the example too.

michaljmatusiak92 commented on docs/metrics/introduction/metric-quantization.md:
````diff
@@ -214,6 +218,14 @@ cpu | quantize to 15m using sum
+Also, you can skip specifying target quantization interval:
+
+```sql
+cpu | quantize using max
````
`metric = CPU_Idle` ?

michaljmatusiak92 commented on docs/metrics/introduction/metric-quantization.md:
````diff
@@ -214,6 +218,14 @@ cpu | quantize to 15m using sum
+cpu | quantize using max
+```
+
+In this case Sumo will automatically determine appropriate interval and quantize to the `max` rollup type.
````
`s/interval/quantisation interval`

michaljmatusiak92 commented on docs/metrics/introduction/metric-quantization.md:
````diff
@@ -268,10 +280,10 @@ If your metric query uses the `quantize` operator without specifying a rollup ty
 
 ```sql
-... | quantize to 5s | parse field=_sourceHost - as cluster,instance | ..
+metric=CPU_Idle | quantize to 5s | parse field=_sourceHost as cluster,instance | avg by cluster
````
I believe the right syntax is:
```
metric=CPU_Idle | quantize to 5s | parse field=_sourceHost * as cluster | avg by cluster
```

michaljmatusiak92 commented on docs/metrics/metrics-operators/quantize.md:
````diff
@@ -56,3 +57,11 @@ The `quantize` clause in this metric query sets the time bucket size to 10 minut
+In this example we omit the `to INTERVAL` part of the query and let Sumo determine appropriate quantization interval according to [How sumo chooses rollup table and quantization interval](../introduction/metric-quantization.md#how-sumo-chooses-rollup-table-and-quantization-interval). `sum` rollup type will be used to aggregate the metrics in each time bucket.
+
+```sql
+metric=CPU_User cluster=kafka | quantize using sum
````
I this case the `sum` of `CPU_User` has a little sense, how about using `avg` in the example?

betenkowski replied:
`avg` is the default but `max` could work better.

aszczepaniksumo replied:
`avg` does not make much sense too as it is the default. I will use non-default `max` which should be more realistic, also for other examples in this doc.

michaljmatusiak92 reviewed (commented):
some minors left, lgtm once applied

betenkowski commented on docs/metrics/metrics-operators/quantize.md:
```diff
@@ -11,27 +11,28 @@ You can specify:
 * `INTERVAL` is the duration over which you want to quantize the metrics, in seconds (`s`) , minutes (`m`), hours (`h`), or days (`d`).
 * `ROLLUP` is  `avg, min, max, sum`, or `count`.
 * `drop last` causes the last time bucket to be dropped, if the end of that bucket is after the end of the query time range.
+* At least one of the `to INTERVAL` or `using ROLLUP` clauses needs to be present.
```
You haven't changed this but worth fixing anyway: extra space before comma in line 22.

aszczepaniksumo commented:
I applied all the remarks 🙂

michaljmatusiak92 commented on docs/metrics/metrics-operators/eval.md:
````diff
@@ -32,18 +31,18 @@ metrics query | eval expr([REDUCER BOOLEAN EXPRESSION | _value] [_granularity])
 
 ```
-_sourceCategory=ApacheHttpServer metrics=cpu_idle | eval _value * 100
+_sourceCategory=ApacheHttpServer metric=CPU_Idle | eval _value * 100
````
I'm wondering if  adding `_sourceCategory=ApacheHttpServer` selector actually adds value to the example - looks like this is a one-off in the examples provided

kimsauce commented on docs/metrics/introduction/metric-quantization.md:
````diff
@@ -200,20 +200,32 @@ The table below shows how Sumo Logic selects a quantization interval based on qu
+metric=CPU_Idle | quantize using max
+```
+
+In this case Sumo will automatically determine appropriate quantization interval and quantize to the `max` rollup type.
````
```suggestion
In this case, Sumo Logic will automatically determine appropriate quantization interval and quantize to the `max` rollup type.
```
